File: backend/local-api-module/src/main/java/org/vyatsu/localApiModule/entity/item/Order.java
package org.vyatsu.localApiModule.entity.item;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Entity
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "id_item", nullable = false)
    private Item item;

    @Column(name = "price", nullable = false)
    private double price;

    @Column(name = "srid", nullable = false)
    private String srid;

    @Column(name = "order_type")
    private String orderType;

    @Column(name = "is_cancel")
    private Boolean isCancel;

    @Column(name = "is_refund", nullable = false)
    private Boolean isRefund;

    @Column(name = "taked_at", columnDefinition = "TIMESTAMP")
    private LocalDateTime takedAt;

    @Column(name = "created_at", columnDefinition = "TIMESTAMP NOT NULL")
    private LocalDateTime createdAt;
}

File: app/src/main/java/com/bytebusters/android/mpsystem/entity/item/Order.kt
package com.bytebusters.android.mpsystem.entity.item

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.time.LocalDateTime

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Data
@Entity
@Table(name = "orders")
class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private val id: Long? = null

    @ManyToOne
    @JoinColumn(name = "id_item", nullable = false)
    private val item: Item? = null

    @Column(name = "price", nullable = false)
    private val price = 0.0

    @Column(name = "taked_at", columnDefinition = "TIMESTAMP NOT NULL", nullable = false)
    private val takedAt: LocalDateTime? = null

    @Column(name = "created_at", columnDefinition = "TIMESTAMP NOT NULL")
    private val createdAt: LocalDateTime? = null
}
